src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredToArmProcessorImpl.java

```diff
@@ -146,17 +147,20 @@ private void processPendingUnstructured() {
 
             String filename = generateFilename(armExternalObjectDirectory);
 
-            boolean copyRawDataToArmSuccessful = copyRawDataToArm(unstructuredExternalObjectDirectory, armExternalObjectDirectory,
-                                                                  filename, failedStatus
+            boolean copyRawDataToArmSuccessful = copyRawDataToArm(unstructuredExternalObjectDirectory,
+                                                                  armExternalObjectDirectory,
+                                                                  filename,
+                                                                  failedStatus
             );
-            if (copyRawDataToArmSuccessful) {
-                generateAndCopyMetadataToArm(armExternalObjectDirectory);
+            if (copyRawDataToArmSuccessful && generateAndCopyMetadataToArm(armExternalObjectDirectory)) {
+                armExternalObjectDirectory.setStatus(armStatuses.get(ARM_DROP_ZONE));
+                externalObjectDirectoryRepository.saveAndFlush(armExternalObjectDirectory);
             }
         }
     }
 
-    private void generateAndCopyMetadataToArm(ExternalObjectDirectoryEntity armExternalObjectDirectory) {
-
+    private boolean generateAndCopyMetadataToArm(ExternalObjectDirectoryEntity armExternalObjectDirectory) {
+        boolean copiedManifestToArm = false;
```

Comment:
It's preferable to avoid setting bool flags like this, as the outcome of the method can be hard to read for the branching cases. Also bugs are more likely as this non-final field may be changed multiple times before the method returns later.

Consider refactoring to remove this flag in favour of using early returns.